Pod/Classes/SwitchViewComponent/ARSwitchViewManager.m

```diff
@@ -2,33 +2,47 @@
 
 #import <Extraction/ARSwitchView.h>
 #import <React/RCTComponent.h>
+#import <FLKAutoLayout/FLKAutoLayout.h>
 
 
-@interface ARSwitchViewComponent : ARSwitchView <ARSwitchViewDelegate>
+@interface ARSwitchViewComponent : UIView <ARSwitchViewDelegate>
+@property (nonatomic, strong, readwrite) ARSwitchView *switchView;
 @property (nonatomic, strong, readwrite) RCTDirectEventBlock onSelectionChange;
 @end
 
 @implementation ARSwitchViewComponent
 
-- (instancetype)initWithButtonTitles:(NSArray *)titles;
+- (instancetype)init;
 {
-  if ((self = [super initWithButtonTitles:titles])) {
-    self.delegate = self;
-  }
-  return self;
+    if ((self = [super init])) {
+        
+        // We are wrapping ARSwitchView within a UIView.
+        // This is done to intercept React Native re-setting the background color on ARSwitchView directly.
```

Comment:
We found this in `RCTUIManager.h`

```
  // Shadow view is the source of truth for background color this is a little
  // bit counter-intuitive if people try to set background color when setting up
  // the view, but it's the only way that makes sense given our threading model
  UIColor *backgroundColor = shadowView.backgroundColor;

  [self addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry){
    UIView *view = [componentData createViewWithTag:reactTag];
    if (view) {
      [componentData setProps:props forView:view]; // Must be done before bgColor to prevent wrong default
      if ([view respondsToSelector:@selector(setBackgroundColor:)]) {
        ((UIView *)view).backgroundColor = backgroundColor;
      }
```